index.bs

```diff
@@ -137,7 +182,46 @@ Coordinates</a></code> interface of the Geolocation API are the following:
 </div>
 
 To <dfn>construct a <a>Geolocation Sensor</a> object</dfn> the user agent must invoke the
-[=construct a Sensor object=] abstract operation.
+[=construct a Sensor object=] abstract operation for the {{GeolocationSensor}} interface.
+
+### GeolocationSensor.read() ### {#geolocationsensor-read}
+
+The {{GeolocationSensor/read()}} method, when called, must run the following algorithm:
+
+<div algorithm="read">
+<!-- https://dom.spec.whatwg.org/#abortcontroller-api-integration -->
+
+    : input
+    :: |options|, a {{ReadOptions}} object.
+    : output
+    :: |p|, a promise.
+
+    1. Let |p| be a new promise.
+    1. Let |options| be the first argument.
+    1. Let |signal| be the |options|' dictionary member of the same name if present, or null otherwise.
+    1. If |signal|'s <a>aborted flag</a> is set, then reject |p| with an "{{AbortError!!exception}}" {{DOMException}} and return |p|.
+    1. Let |geo| be the result of invoking <a>construct a Geolocation Sensor object</a> with |options|. If this throws a {{DOMException}}, catch it, reject |p| with that {{DOMException}}, and return |p|.
+    1. Invoke |geo|.{{Sensor/start()}}.
+    1. If |signal| is not null, then <a>add the following abort steps</a> to |signal|:
+      1. Invoke |geo|.{{Sensor/stop()}}.
+      1. Reject |p| with an "{{AbortError!!exception}}" {{DOMException}} and abort these steps.
+    1. Run these steps <a>in parallel</a>:
+      1. If <a>notify error</a> is invoked with |geo|, reject |p| with the {{DOMException}} passed as input to <a>notify error</a>.
+      1. If <a>notify new reading</a> is invoked with |geo|, then <a>resolve geolocation promise</a> with |geo| and |p|.
+    1. Return |p|.
+</div>
+
+<p class="note">
+Implementations can reuse the same promise for multiple concurrent calls within the same browsing context if the arguments passed to {{read()}} are the same.
+</p>
+
+To <dfn>resolve geolocation promise</dfn> means to run the following steps:
+  1. Let |reading| be a {{GeolocationSensorReading}}.
+  1. For each |key| → |value| of <a>latest geolocation reading</a>:
+    1. Set |reading|[key] to |value|.
+  1. Resolve |p| with |reading|.
```

Comment:
It's safer to stop sensor before resolve

Reply:
Fixed.